### PR TITLE
One-pager "choosing your connection" to help address points of customer confusion

### PR DESCRIPTION
**Pull request by bgrenon**
Reviewers:

This page is intended to help users make their choice of driver and connection type, with clarification of the various gotchas discussed in various recent slack threads. Please help:

- Validate the [flowchart](https://neon-next-git-bgrenon-pooling-refresh-neondatabase.vercel.app/docs/connect/choose-connection#choosing-your-connection-type-flowchart). 
- Suggest any other [common pitfalls](https://neon-next-git-bgrenon-pooling-refresh-neondatabase.vercel.app/docs/connect/choose-connection#common-pitfalls), pain points, or areas of confusion that you feel we should include here based on your experience troubleshooting with customers.

Preview:
https://neon-next-git-bgrenon-pooling-refresh-neondatabase.vercel.app/docs/connect/choose-connection

### CODE REVIEW

File: content/docs/connect/choose-connection.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+---
+title: Choosing your connection type
+subtitle: Rules of thumb for choosing one connection type over another
+enableTableOfContents: true
+---
+
+When setting up your application’s connection to your Neon Postgres database, you need to make two main choices: the type of driver and the type of connection. This flowchart will guide you through these options.
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:7]:
```suggestion
When setting up your application’s connection to your Neon Postgres database, you need to make two main choices: 

- **Postgres driver type**: Serverless driver vs. TCP-based driver
- **Connection type**: pooled vs. unpooled

This flowchart will guide you through these selections.
```

Comment (danieltprice) [reply content/docs/connect/choose-connection.md:7]: not sure about "direct" vs "unpooled". We're using both in the docs currently, and the Vercel integration defines a "DATABASE_URL_UNPOOLED" variable. wdyt?

File: content/docs/connect/choose-connection.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+## Choosing your connection type: flowchart
+
+![choose your connection type](/docs/connect/choose_connection.png)
+
+## Choosing your connection type: drivers and pooling
+
+### Step 1: Choose your Postgres driver
+
+Your first choice is which driver to use. If working in a serverless environment, use the [Neon Serverless Driver](/docs/serverless/serverless-driver). It's built to handle dynamic workloads with high variability in traffic &#8212; for example, Vercel Edge Functions or Cloudflare Workers.
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:17]:
```suggestion
Your first choice is which driver to use. If working in a serverless environment and connecting from a JavaScript and TypeScript application, consider using the [Neon Serverless Driver](/docs/serverless/serverless-driver).  This is a low-latency Postgres driver that allows you to query data from serverless and edge environments over HTTP or WebSockets instead of TCP. It's built to handle dynamic workloads with high variability in traffic &#8212; for example, Vercel Edge Functions or Cloudflare Workers.
```

File: content/docs/connect/choose-connection.md (hunk 1 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md:7]: Unless it's against industry convention (the terms seem interchangeable from my searches), to me "direct" is clearer. If pooled was the default, so that you have to explicitly disable it, then "unpooled" would make sense since you're making a clear decision to not use pooling.

File: content/docs/connect/choose-connection.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+title: Choosing your connection type
+subtitle: Rules of thumb for choosing one connection type over another
+enableTableOfContents: true
+---
+
+When setting up your application’s connection to your Neon Postgres database, you need to make two main choices: the type of driver and the type of connection. This flowchart will guide you through these options.
+
+## Choosing your connection type: flowchart
+
+![choose your connection type](/docs/connect/choose_connection.png)
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:11]: Not sure about "Is your pool size limited by the default 64 connections?" By default, everyone has a PgBouncer setting of default_pool_size=64. So everyone's pool is limited to 64 unless that setting is changed. 

Maybe the question here is: 

Will an individual Postgres role require more than 64 concurrent connections to a database? 

If the answer is "yes"

"Contact Support to increase your PgBouncer default_pool_size setting."

File: content/docs/connect/choose-connection.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+## Choosing your connection type: drivers and pooling
+
+### Step 1: Choose your Postgres driver
+
+Your first choice is which driver to use. If working in a serverless environment, use the [Neon Serverless Driver](/docs/serverless/serverless-driver). It's built to handle dynamic workloads with high variability in traffic &#8212; for example, Vercel Edge Functions or Cloudflare Workers.
+
+If you don't need serverless, use a traditional TCP-based Postgres driver. For example, if you’re using Node.js with a framework like Next.js, you can add the `pg` client to your dependencies, which acts as the TCP-based Postgres driver.
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:19]:
```suggestion
If you're not connecting from a JavaScript or TypeScript application or you are not developing a serverless application, use a traditional TCP-based Postgres driver. For example, if you’re using Node.js with a framework like Next.js, you can add the `pg` client to your dependencies, which serves as the Postgres driver for TCP connections.
```

File: content/docs/connect/choose-connection.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+### Step 2: If serverless, choose HTTP or WebSocket connections
+
+Neon offers two types of serverless connections: HTTP and WebSocket.
+
+- **Use HTTP for**: One-shot queries and short-lived operations where connection setup and teardown costs are minimal.
+- **Use WebSocket for**: [Transactions](/docs/postgresql/query-reference#transactions) and operations requiring persistent connections over a session. WebSocket connections provide lower latency for multiple queries in the same session.
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:31]:
```suggestion
- **Use WebSocket for**: [Transactions](/docs/postgresql/query-reference#transactions) and operations requiring persistent connections over a session. WebSocket connections provide lower latency for multiple queries in the same session and are fully compatible with [node-postgres](https://node-postgres.com/).
```

Comment (danieltprice) [reply content/docs/connect/choose-connection.md:31]: Not sure about the "provide lower latency for multiple queries in the same session". Lower than HTTP or lower than TCP connections? HTTP connections support batched queries (a batch of one-shot queries), but not queries within a transaction. so this statement might be confusing.

File: content/docs/connect/choose-connection.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+<Admonition type="note">
+We are working on automatic switching between HTTP and WebSocket to as needed. Check our [roadmap](/docs/introduction/roadmap) to see what's coming soon and our Friday [Changelog](/docs/changelog) for the features-of-the-week.
+</Admonition>
+
+### Step 3: Next, choose your connection type: direct or pooled
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:37]: "direct? or "unpooled"?

File: content/docs/connect/choose-connection.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+You then need to decide whether to use direct connections or pooled connections using PgBouncer:
+
+- **In general, use pooled connections whenever you can**: Pooled connections can efficiently manage high numbers of concurrent connections, supporting up to 10,000 connection. This 10,000 connection limit is most useful for serverless applications and application-side connection pools that have many open connections, but infrequent and/or short transactions.
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:41]:
```suggestion
- **In general, use pooled connections whenever you can**: Pooled connections can efficiently manage high numbers of concurrent connections, supporting up to 10,000 concurrent connections. This 10,000 connection limit is most useful for serverless applications and application-side connection pools that have many open connections, but infrequent and/or short transactions.
```

File: content/docs/connect/choose-connection.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+- **Use direct (unpooled) connections if you need persistent connections**: If your application is focused mainly on tasks like migrations or administrative operations that require stable and long-lived connections, use an unpooled connection.
+
+<Admonition type="note">
+The use of connection pooling is not a magic bullet. PgBouncer has a `default_pool_size` of 64, meaning it can handle only 64 direct connections to the database at any one time. When your application connects to PgBouncer, it joins a large pool, but only 64 connections are processed at a time. Long operations can cause these connections to time out or close before tasks are completed.
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:45]:
```suggestion
Connection pooling is not a magic bullet. It can manage 10,000 concurrent connections from your application to PgBouncer, but PgBouncer's default_pool_size setting is set to 64 by default, which means connections from PgBouncer to Postgres are limited to 64 for each user-database pair. For example, Postgres user `alex` can hold up to 64 connections to a single database at one time.
```

File: content/docs/connect/choose-connection.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+- [Neon Serverless Driver](/docs/serverless/serverless-driver)
+- [Connection pooling](/docs/connect/connection-pooling)
+
+## Common Pitfalls
+
+Here are some key points to help you navigate potential issues.
+| Issue | Description |
+|----------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Double Pooling | Avoid using client-side pooling if you're using PgBouncer on the Neon side. Just let Neon handle the pooling to prevent retaining unused connections on the client side. If you must use client-side pooling, make sure it releases connections back to the pool early enough to avoid conflicts with PgBouncer. |
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:58]:
```suggestion
| Double Pooling | Avoid using client-side pooling if you're using a pooled Neon connection (supported by PgBouncer). Just let Neon handle the pooling to prevent retaining unused connections on the client side. If you must use client-side pooling, make sure connections are released back to the client-side pool early enough to avoid conflicts with PgBouncer. |
```

File: content/docs/connect/choose-connection.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+| Understanding Limits | Don't confuse `max_connections` with `default_pool_size`.<br /><br />`max_connections` is the maximum number of concurrent connections allowed by PostgreSQL and is determined by your compute size.<br /><br />`default_pool_size` is the maximum number of connections PgBouncer will pool per user/database pair, which is set to 64 by default.<br /><br />Simply increasing your compute to get more `max_connections` may not improve performance if the bottleneck is actually on your `default_pool_size`. To increase your `default_pool_size`, contact support. |
+
+## Connection types for typical applications
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:61]: Not sure about this section. This kind of statement seems questionable:
"direct connections for inventory updates"
I'd leave it out unless we can validate with an authoritative source.

File: content/docs/connect/choose-connection.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+
+- **E-commerce Platforms:** High concurrency and rapid read/write operations. Use pooled connections for handling high traffic and direct connections for inventory updates.
+- **Content Management Systems (CMS):** Frequent read operations and moderate write operations. Use direct connections for content updates and pooled connections for concurrent reads.
+- **Financial Applications:** High concurrency and secure transactions. Use direct connections for stability and security, and pooled connections for high-frequency reads.
+- **Analytics and Reporting Tools:** Long-running queries and high data throughput. Use direct connections for report generation and pooled connections for frequent data access.
+- **Serverless Applications:** Rapid scaling and short-lived connections. Use the serverless driver with HTTP for one-shot queries and WebSocket for persistent transactions.
+
+## Configuration Examples
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:69]: I think we're better off pointing to the relevant content for examples. 

The "Configure the connection for HTTP or WebSocket" only shows a WebSocket example, for instance, and there are many other examples for both HTTP and WebSockets in the Serverles driver topic:
https://neon.tech/docs/serverless/serverless-driver

File: content/docs/connect/choose-connection.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+
+   ```javascript
+   client.query('SELECT * FROM users', (err, res) => {
+     if (err) throw err;
+     console.log(res.rows);
+     client.end();
+   });
+   ```
+
+## Monitoring
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:149]: Maybe monitoring connections deserves its own topic.

File: content/docs/connect/choose-connection.md [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+
+- **Monitoring Dashboard**: Use the [Monitoring Dashboard](docs/introduction/monitoring-page) in the Neon Console to monitor connection metrics, compute usage, and database health.
+
+- **External tools**: Use external tools like PgHero and pgAdmin to monitor your database. See [Monitoring Neon with external tools](docs/introduction/monitor-external-tools) for information about setting up and using these tools.
+
+## Table summarizing your options
+
+Here is a table summarizing the options we've walked through on this page:
+
+| Feature                   | Direct Connections                                                                                   | Pooled Connections                                                                   | Serverless Driver (HTTP)                 | Serverless Driver (WebSocket)                 |
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:163]: The table is not showing up correctly in the output.

File: content/docs/connect/choose-connection.md (hunk 3 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md:16]: Modify "unpooled" in the rest of the docs before finishing this PR, pending confirmation.

File: content/docs/connect/choose-connection.md (hunk 5 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md:31]: double-check "lower latency for multiple queries"

File: content/docs/connect/choose-connection.md (hunk 8 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md:45]: Could it be a combination of connections held open for too long, or is it also volume?

File: content/docs/connect/choose-connection.md (hunk 3 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md:16]: Modified the flowchart. Flowchart seems busy, but maybe PixelPoint can make it feel simpler.

File: content/docs/connect/choose-connection.md (hunk 5 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md:31]: I modified to just use the same text as the serverless page: unless it can be simplified, why re-invent the wheel

File: content/docs/connect/choose-connection.md [hunk 14]
```diff
@@ -0,0 +1,163 @@
+---
+title: Choosing your connection type
```
[the remark this answers is not in the document]
Comment (bgrenon) [reply content/docs/connect/choose-connection.md:2]: Done.

File: content/docs/connect/choose-connection.md (hunk 11 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md]: Removed to stash file for now

File: content/docs/connect/choose-connection.md (hunk 13 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md:233]: fixed

File: content/docs/connect/choose-connection.md (hunk 12 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md]: Moved to stash file for now

File: content/docs/connect/choose-connection.md (hunk 10 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md]: moved to stash file for now

File: content/docs/connect/choose-connection.md [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+- **In general, use pooled connections whenever you can**
+
+  Pooled connections can efficiently manage high numbers of concurrent connections, supporting up to 10,000 concurrent connections. This 10,000 connection limit is most useful for serverless applications and application-side connection pools that have many open connections, but infrequent and/or short transactions.
+
+- **Use direct (unpooled) connections if you need persistent connections**
+
+  If your application is focused mainly on tasks like migrations or administrative operations that require stable and long-lived connections, use an unpooled connection.
+
+<Admonition type="note">
+Connection pooling is not a magic bullet. PgBouncer can manage 10,000 concurrent connections from your application to PgBouncer, but PgBouncer's default_pool_size setting is set to 64 by default, which means connections from PgBouncer to Postgres are limited to 64 for each user-database pair. For example, Postgres user `alex` can hold up to 64 connections to a single database at one time.
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:61]:
```suggestion
Connection pooling is not a magic bullet. PgBouncer can manage 10,000 concurrent connections from your application to PgBouncer, but PgBouncer's `default_pool_size` setting is set to 64 by default, which means connections from PgBouncer to Postgres are limited to 64 for each user-database pair. For example, Postgres user `alex` can hold up to 64 connections to a single database at one time.
```

File: content/docs/connect/choose-connection.md [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+
+Here's an example of getting a direct connection string from the Neon CLI:
+
+```bash shouldWrap
+neonctl connection-string [branch_name]
+
+postgres://alex:AbC123dEf@ep-cool-darkness-123456.us-east-2.aws.neon.tech/dbname?sslmode=require
+```
+
+For more detail, see [How to use connection pooling](/docs/connect/connection-pooling#how-to-use-connection-pooling)
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:124]:
```suggestion
For more detail, see [How to use connection pooling](/docs/connect/connection-pooling#how-to-use-connection-pooling).
```

File: content/docs/connect/choose-connection.md [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+You can use standard Postgres client libraries or drivers. Neon is fully compatible with Postgres, so any application or utility that works with Postgres should work with Neon. Consult the integration guide for your particular language or framework for the right client for your needs:
+
+- [Framework Quickstarts](/docs/get-started-with-neon/frameworks)
+- [Language Quickstarts](/docs/get-started-with-neon/languages)
+
+### Configuring the connection
+
+Setting up a direct or pooled connection is usually a matter of choosing the appropriate connection string and adding it to your application's `.env` file.
+
+You can get your connection string from the Neon console or via CLI.
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:104]:
```suggestion
You can get your connection string from the [Neon Console](/docs/connect/connect-from-any-app) or via CLI.
```

File: content/docs/connect/choose-connection.md [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+| **Use Case**              | Migrations, admin tasks requiring stable connections                                                 | High number of concurrent connections, efficient resource management                 | One-shot queries, short-lived operations | Transactions requiring persistent connections |
+| **Connection Management** | Persistent connections                                                                               | Pooled per transaction                                                               | Rapid open/close connections             | Rapid open/close connections                  |
+| **Scalability**           | Limited by `max_connections` tied to [compute size](/docs/manage/endpoints#how-to-size-your-compute) | Default pool size of 64 connections per user/database, can be increased with support | Automatically scales                     | Automatically scales                          |
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:134]:
```suggestion
| **Scalability**           | Limited by `max_connections` tied to [compute size](/docs/manage/endpoints#how-to-size-your-compute) | Default pool size of 64 connections per user/database pair, can be increased upon request | Automatically scales                     | Automatically scales                          |
```

File: content/docs/connect/choose-connection.md [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+| ------------------------- | ---------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------ | ---------------------------------------- | --------------------------------------------- |
+| **Use Case**              | Migrations, admin tasks requiring stable connections                                                 | High number of concurrent connections, efficient resource management                 | One-shot queries, short-lived operations | Transactions requiring persistent connections |
+| **Connection Management** | Persistent connections                                                                               | Pooled per transaction                                                               | Rapid open/close connections             | Rapid open/close connections                  |
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:133]:
```suggestion
| **Connection Management** | Persistent connections                                                                               | Served from pool                                                               | Rapid open/close connections             | Rapid open/close connections                  |
```

File: content/docs/connect/choose-connection.md [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+postgres://alex:AbC123dEf@ep-cool-darkness-123456.us-east-2.aws.neon.tech/dbname?sslmode=require
+```
+
+For more detail, see [How to use connection pooling](/docs/connect/connection-pooling#how-to-use-connection-pooling)
+
+## Table summarizing your options
+
+Here is a table summarizing the options we've walked through on this page:
+
+| Feature                   | Direct Connections                                                                                   | Pooled Connections                                                                   | Serverless Driver (HTTP)                 | Serverless Driver (WebSocket)                 |
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:130]:
```suggestion
|                    | Direct Connections                                                                                   | Pooled Connections                                                                   | Serverless Driver (HTTP)                 | Serverless Driver (WebSocket)                 |
```

File: content/docs/connect/choose-connection.md [hunk 21]
```diff
@@ -0,0 +1,133 @@
...
+When setting up your application’s connection to your Neon Postgres database, you need to make two main choices:
+
+- **The right driver for your deployment** &#8212; Neon Serverless driver or a TCP-based driver
+- **The right connection type for your traffic** &#8212; pooled connections or direct connections
+
+This flowchart will guide you through these selections.
+
+## Choosing your connection type: flowchart
+
+![choose your connection type](/docs/connect/choose_connection.png)
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:16]: "per user role" should be "per user-database pair"

File: content/docs/connect/choose-connection.md [hunk 22]
```diff
@@ -0,0 +1,133 @@
...
+
+## Choosing your connection type: drivers and pooling
+
+### Your first choice is which driver to use
+
+- **Serverless**
+
+  If working in a serverless environment and connecting from a JavaScript and TypeScript application, we recommend using the [Neon Serverless Driver](/docs/serverless/serverless-driver). It handles dynamic workloads with high variability in traffic &#8212; for example, Vercel Edge Functions or Cloudflare Workers.
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:24]:
```suggestion
  If working in a serverless environment and connecting from a JavaScript or TypeScript application, we recommend using the [Neon Serverless Driver](/docs/serverless/serverless-driver). It handles dynamic workloads with high variability in traffic &#8212; for example, Vercel Edge Functions or Cloudflare Workers.
```

File: content/docs/connect/choose-connection.md [hunk 23]
```diff
@@ -0,0 +1,133 @@
...
+- **In general, use pooled connections whenever you can**
+
+  Pooled connections can efficiently manage high numbers of concurrent client connections, up to 10,000. This 10K ceiling works best for serverless applications and application-side connection pools that have many open connections, but infrequent and/or short transactions.
+
+- **Use direct (unpooled) connections if you need persistent connections**
+
+  If your application is focused mainly on tasks like migrations or administrative operations that require stable and long-lived connections, use an unpooled connection.
+
+<Admonition type="note">
+Connection pooling is not a magic bullet. PgBouncer can keep many client connections open (up to 10,000) concurrently, but only a limited number of these can be actively querying the PostgreSQL server at any given time: 64 active backend transactions per user-database pair, as set by the PgBouncer's `default_pool_size` setting. For example, the Postgres user `alex` can hold up to 64 backend connections to a single database at one time.
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:59]:
```suggestion
Connection pooling is not a magic bullet. PgBouncer can keep many client connections open (up to 10,000) concurrently, but only a limited number of these can be actively querying the Postgres server at any given time: 64 active backend transactions per user-database pair, as set by the PgBouncer's `default_pool_size` setting. For example, the Postgres user `alex` can hold up to 64 backend connections to a single database at one time.
```

File: content/docs/connect/choose-connection.md [hunk 24]
```diff
@@ -0,0 +1,133 @@
...
+
+- [Neon Serverless Driver](/docs/serverless/serverless-driver)
+- [Connection pooling](/docs/connect/connection-pooling)
+
+## Common Pitfalls
+
+Here are some key points to help you navigate potential issues.
+| Issue | Description |
+|----------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Double pooling | Avoid using client-side pooling if you're using a pooled Neon connection (supported by PgBouncer). Just let Neon handle the pooling to prevent retaining unused connections on the client side. If you must use client-side pooling, make sure connections are released back to the client-side pool early enough to avoid conflicts with PgBouncer. |
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:72]: We mentioned above that the 10K limit works well for application-side connection pooling, but here we say to avoid using client-side pooling with a pooled connection.

File: content/docs/connect/choose-connection.md [hunk 25]
```diff
@@ -0,0 +1,133 @@
...
+| Understanding limits | Don't confuse `max_connections` with `default_pool_size`.<br /><br />`max_connections` is the maximum number of concurrent connections allowed by Postgres and is determined by your [Neon compute size](/docs/connect/connection-pooling#connection-limits-without-connection-pooling).<br /><br />`default_pool_size` is the maximum number of backend connections or transactions that PgBouncer supports per user/database pair, which is set to 64 by default.<br /><br />Simply increasing your compute to get more `max_connections` may not improve performance if the bottleneck is actually on your `default_pool_size`. To increase your `default_pool_size`, contact [Support](/docs/introduction/support). |
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:73]: not sure about "backend connections".

File: content/docs/connect/choose-connection.md [hunk 26]
```diff
@@ -0,0 +1,133 @@
...
+| **Use Case**              | Migrations, admin tasks requiring stable connections                                                 | High number of concurrent client connections, efficient resource management                                     | One-shot queries, short-lived operations | Transactions requiring persistent connections |
+| **Connection Management** | Persistent connections                                                                               | Served from pool                                                                                                | Rapid open/close connections             | Rapid open/close connections                  |
+| **Scalability**           | Limited by `max_connections` tied to [compute size](/docs/manage/endpoints#how-to-size-your-compute) | Up to 10,000 client connections; 64 backend transactions per user/database pair (can be increased upon request) | Automatically scales                     | Automatically scales                          |
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:132]: Not sure about "backend"

File: content/docs/connect/choose-connection.md [hunk 27]
```diff
@@ -0,0 +1,133 @@
...
+| ------------------------- | ---------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------- | ---------------------------------------- | --------------------------------------------- |
+| **Use Case**              | Migrations, admin tasks requiring stable connections                                                 | High number of concurrent client connections, efficient resource management                                     | One-shot queries, short-lived operations | Transactions requiring persistent connections |
+| **Connection Management** | Persistent connections                                                                               | Served from pool                                                                                                | Rapid open/close connections             | Rapid open/close connections                  |
```
Comment (danieltprice) [inline content/docs/connect/choose-connection.md:131]: Not sure what "Connection Management" is

File: content/docs/connect/choose-connection.md (hunk 25 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md:73]: I think it helps clarify: 
- client connections = open connections between the application and pgbouncer
- backend connections = active transactions between pgbouncer and the postgres 

I think some of the customer confusion might come from where we describe both sides of the pgbouncer handoff as "concurrent connections". Using "client" and "backend" where we can might help. The terms seemsstandard in various postgres docs/posts, and we should probably update our pooling docs to define and use those those terms, and use them in new diagrams as well.

So basically, double down. :)
WDYT?

Comment (danieltprice) [reply content/docs/connect/choose-connection.md:73]: ok! This should help -> "define and use those those terms, and use them in new diagrams as well"

File: content/docs/connect/choose-connection.md (hunk 24 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md:72]: Ok, I'll clarify this. The thing to avoid is pooling that's managed within the client library, doubled up with pooling provided by the application's connections to pgbouncer: if you want to use both (ie double-pooling) you need to configure the client-side pooling more carefully.

File: content/docs/connect/choose-connection.md (hunk 25 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md:74]: For now, added some small clarifications throughout.

**Review by danieltprice — APPROVED** [review]

File: content/docs/connect/choose-connection.md (hunk 26 above)
Comment (bgrenon) [reply content/docs/connect/choose-connection.md]: clarified
